packages/edit-widgets/src/register-legacy-widget-variations.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import { subscribe, select, dispatch } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as blocksStore } from '@wordpress/blocks';
+
+export default function registerLegacyWidgetVariations( settings ) {
+	const unsubscribe = subscribe( () => {
+		const hiddenIds =
+			settings?.widgetTypesToHideFromLegacyWidgetBlock ?? [];
+		const widgetTypes = select( coreStore )
+			.getWidgetTypes( { per_page: -1 } )
+			?.filter( ( widgetType ) => ! hiddenIds.includes( widgetType.id ) );
```

Comment:
Here too.
```suggestion
			?.filter?.( ( widgetType ) => ! hiddenIds.includes( widgetType.id ) );
```

Reply:
Same again. If `getWidgetTypes()` returns something truthy then we know it will have `filter`.